File: backend/models/spell.model.js
const mongoose = require("mongoose");
const schema = mongoose.Schema;

const spellSchema = schema({
    spell_id: {type: Number, required: true},
    name: {type: String, required: true},
    image: {type: String, required: true},
});

const EldenBuild = mongoose.model("spells", spellSchema);

module.exports = EldenBuild;

File: lib/models/spell_model.dart
class Spell {
  int spell_id;
  String name;
  String image;

  Spell(
      { required this.spell_id,
      required this.name,
      required this.image});

  Spell.fromJson(Map<String, dynamic> json)
      : spell_id = json['spell_id'],
        name = json['name'],
        image = json['image'];

  Map<String, dynamic> toJson() {
    return {
      'spell_id': spell_id,
      'name': name,
      'image': image,
    };
  }

  String showSpell() {
    return "$name\n$spell_id\n$image\n";
  }
}
